File: Web.Admin/src/api/services/Admin/employee/models/EditEmployeeInput.ts
import type { EmployeeOrgModel } from "./EmployeeOrgModel";
import type { EmployeeRoleModel } from "./EmployeeRoleModel";
import type { GenderEnum } from "@/api/enums/GenderEnum";

/**
 * 编辑职员输入
 */
export interface EditEmployeeInput {
	/**
	 * 职员Id
	 */
	employeeId?: number;
	/**
	 * 姓名
	 */
	employeeName?: string;
	/**
	 * 手机
	 */
	mobile?: string;
	/**
	 * 邮箱
	 */
	email?: string;
	/**
	 * 
	 */
	sex?: GenderEnum;
	/**
	 * 证件照
	 */
	idPhoto?: string;
	/**
	 * 入职日期
	 */
	entryDate?: string;
	/**
	 * 备注
	 */
	remark?: string;
	/**
	 * 机构信息
	 */
	orgList?: EmployeeOrgModel[];
	/**
	 * 角色信息
	 */
	roleList?: EmployeeRoleModel[];
	/**
	 * 
	 */
	rowVersion?: number;
}

File: Api.Server/src/Admin.Service/Employee/Dto/EditEmployeeInput.cs
// Copyright © 2018-Present 小方
// SPDX-License-Identifier: Apache-2.0
// 
// 本文件依据 Apache License 2.0 授权，完整条款见仓库根目录 LICENSE。
// 本软件按“原样”提供，相关免责声明及责任限制以许可证及适用法律为准。
// 版权来源、合法使用与二次开发责任说明见仓库根目录 README.md。

using Fast.Admin.Domain;

namespace Fast.Admin.Service.Employee.Dto;

/// <summary>
/// 编辑职员输入
/// </summary>
public class EditEmployeeInput : UpdateVersionInput
{
    /// <summary>
    /// 职员Id
    /// </summary>
    [LongRequired(ErrorMessage = "职员Id不能为空")]
    public long EmployeeId { get; set; }

    /// <summary>
    /// 姓名
    /// </summary>
    [StringRequired(ErrorMessage = "姓名不能为空")]
    public string EmployeeName { get; set; }

    /// <summary>
    /// 手机
    /// </summary>
    [StringRequired(ErrorMessage = "手机不能为空")]
    [RegularExpression(RegexConst.Mobile, ErrorMessage = "手机格式不正确")]
    public string Mobile { get; set; }

    /// <summary>
    /// 邮箱
    /// </summary>
    [StringRequired(ErrorMessage = "邮箱不能为空")]
    [RegularExpression(RegexConst.EmailAddress, ErrorMessage = "邮箱格式不正确")]
    public string Email { get; set; }

    /// <summary>
    /// 性别
    /// </summary>
    [EnumRequired(ErrorMessage = "性别不能为空", AllowZero = true)]
    public GenderEnum Sex { get; set; }

    /// <summary>
    /// 证件照
    /// </summary>
    public string IdPhoto { get; set; }

    /// <summary>
    /// 入职日期
    /// </summary>
    [DateTimeRequired(ErrorMessage = "入职日期不能为空")]
    public DateTime EntryDate { get; set; }

    /// <summary>
    /// 备注
    /// </summary>
    public string Remark { get; set; }

    /// <summary>
    /// 机构信息
    /// </summary>
    public List<EmployeeOrgModel> OrgList { get; set; }

    /// <summary>
    /// 角色信息
    /// </summary>
    public List<EmployeeRoleModel> RoleList { get; set; }
}
